### PR TITLE
HARP-6096: Add fast conversions between the WEB Mercator projections.

### PR DESCRIPTION
**Pull request by robertoraggi**
Also, this change places the "WEB Mercator"-projection variants
currently supported by @here/harp-geoutils in the same file. This is
needed to avoid possible circular dependencies (e.g reprojectPoint()
needs to access both webMercatorProjection and mercatorProjection).

Placing both variants in the same file has the nice side-effect to
remove the need to export the class MercatorProjection.

Signed-off-by: Roberto Raggi <roberto.raggi@here.com>

### CODE REVIEW

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts [hunk 1]
```diff
@@ -183,9 +183,131 @@ export class MercatorProjection extends Projection {
...
+        worldPos: Vector3Like,
+        result?: Vector3Like
+    ): Vector3Like {
+        // this implementation of [[reprojectPoint]] supports both
+        // [[WebMercatorProjection]] and [[MercatorProjection]]. The only
+        // difference betweeen these two variants of WEB Mercator
+        // is in the orientation of the Y axis, so we just flip Y coordinates
+        // when reprojecting between them.
+        if (
+            sourceProjection !== this &&
```
Comment (ninok) [inline @here/harp-geoutils/lib/projection/MercatorProjection.ts]: if sourceProjection === this we should return the input point but instead you call super.reprojectPoint ?!

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts [hunk 2]
```diff
@@ -183,9 +183,131 @@ export class MercatorProjection extends Projection {
...
+            (sourceProjection === webMercatorProjection || sourceProjection === mercatorProjection)
```
Comment (ninok) [inline @here/harp-geoutils/lib/projection/MercatorProjection.ts]: If source is mercator (and this is also mercator) we should not flip the y axis

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts (hunk 1 above)
Comment (robertoraggi) [reply @here/harp-geoutils/lib/projection/MercatorProjection.ts]: when sourceProjection === this we call the base implementation that already does what you expect.

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts (hunk 2 above)
Comment (robertoraggi) [reply @here/harp-geoutils/lib/projection/MercatorProjection.ts]: yes, and that is captured by "sourceProjection !== this"

**Pull request merged**